src/shared/utilities/timeoutUtils.ts

```diff
@@ -0,0 +1,50 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+/**
+ * Timeout coTimeoutbject that can handle both cancellation token- and time limit-style timeout situations.
+ * @param timeoutLength Length of timeout duration (in ms)
+ */
+export class Timeout {
+    private readonly _startTime: number
+    private readonly _endTime: number
+    private readonly _timer: Promise<boolean>
+    public constructor(timeoutLength: number) {
+        this._startTime = new Date().getTime()
+        this._endTime = this._startTime + timeoutLength
+        this._timer = new Promise<boolean>((resolve) => {
+            setTimeout(() => resolve(false), timeoutLength)
+        })
+    }
+
+    /**
+     * Returns the amount of time left from the initialization of time Timeout object and with the timeoutLength
+     * Bottoms out at 0
+     */
+    public get remainingTime(): number {
+        const curr = new Date().getTime()
+
+        return (this._endTime - curr > 0 ? this._endTime - curr : 0)
+    }
+
+    /**
+     * Returns a promise that times out after timeoutLength ms have passed since Timeout object initialization
+     * Use this in Promise.race() calls in order to time out awaited functions
+     * Once this timer has finished, cannot be restarted
+     */
+    public get timer(): Promise<boolean> {
+        return this._timer
+    }
+
+    /**
+     * Returns the elapsed time from the initial Timeout object creation
+     * Useful for telemetry reporting!
```

Comment:
If the timer has finished, should elapsedTime only return the elapsed time of the timer instead of continuing to measure?

Reply:
The idea is that this can be used to kick off a timer that can be used for metrics later, even for things outside the scope of the timeout object.

This object was originally called `InvokeTimer` and was scoped only to the `localLambdaRunner` file; I'm open to changing this functionality or changing the name to better suit this functionality.

Reply:
I wrote that comment before seeing where it is used in the PR. I think it is valid, and if we need to know the "duration of the actual timer" that can become a separate property with an appropriate name. 

No action required here.